info-articles/Mapping in solidity - issue #19.md/article.md

```diff
@@ -0,0 +1,414 @@
+```
+// SPDX-License-Identifier : MIT
+ 
+ pragma solidity ^0.8.17;
+ 
+  contract MappingLesson{
+   
+    mapping(address => uint) public theBalanceOf;
+ 
+    function settingMapping() external{
+     theBalanceOf[msg.sender] = 3 ether;
+    }
+   
+    function gettingMapping() external view{
+      // for set address
+      uint bal = theBalanceOf[msg.sender];
+ 
+      // for unset address
+      uint bal13 = theBalanceOf[address(13)];
+    }
+ 
+    function updatingMapping() external{
+       theBalanceOf[msg.sender] += 13 ether;
+       // this will be the result of 13 + 3
+       // 16 total ethers
+    }
+ 
+    function clearingMapping() external{
+      delete theBalanceOf[address(13)];
+      delete theBalanceOf[msg.sender];
+    }
+ 
+ 
+  }
+```
+Now, let us break these processes down.
+
+### Setting Mapping
+Setting the value for mapping is quite straightforward; place the mapping beside an address that receives ether and assign it to a value.
+```
+function settingMapping() external{
+     theBalanceOf[msg.sender] = 3 ether;
+    }
+```
+### Getting Mapping
+You can fetch the value of a mapping. But to do that, you need to create a variable and assign it to the address mapping.
+
+At this point, remember that you can get the value of both the mappings you have set and those you have not.
+
+In the former case, Solidity will return the value you set to it. But for unset mappings, if the slot does not have a value yet, it will return the default value of uint, which is zero.
+```
+ function gettingMapping() external view{
+      // for set address
+      uint bal = theBalanceOf[msg.sender];
+ 
+      // for unset address
+      uint bal13 = theBalanceOf[address(13)];
+    }
+```
+### Updating Mapping
+You may need to work with the current state of mapping, whether you would like to increase or decrease it. Thus, you have to update the mapping.
+
+If you increase the value of the mapping during the updating process, be aware that the incrementaion will be an addition to the current value.
+
+For instance, if the mapping value is 3 and you update it to 13, the total value will now be 16.
+```
+function updatingMapping() external{
+       theBalanceOf[msg.sender] += 13 ether;
+```
+But you need to note an important fact: you cannot update the mapping of a value you have not set.
+
+### Clearing Mapping
+Perhaps you want to get rid of the value in a mapping, you can clear it by using the `delete keyword`.
+```
+ function clearingMapping() external{
+      delete theBalanceOf[address(13)];
+      delete theBalanceOf[msg.sender];
+```
+
+## How to Work with Arrays in Mapping
+There is a slight variation when you have strings of arrays as the value type of your mapping, especially when it is a dynamic array. In such instances, be mindful of your parameters.
+```
+ contract MappingArrays{

```

Comment:
This code snippet has a missing closing parenthesis **}** in the end

Reply:
```suggestion
 contract MappingArrays{
```suggestion
 contract MappingArrays{

```
```

Reply:
Added